Install with "react": "^18.0.0" not working

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: apperra@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.9.0 || ^17.0.0" from react-currency-input-field@3.6.4
npm ERR! node_modules/react-currency-input-field
npm ERR!   react-currency-input-field@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\MarcStöckli\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MarcStöckli\AppData\Local\npm-cache\_logs\2022-03-31T21_00_46_992Z-debug.log

--- a/package.json
+++ b/package.json
@@ -84,7 +84,7 @@
     "typescript": "^4.2.4"
   },
   "peerDependencies": {
-    "react": "^16.9.0 || ^17.0.0"
+    "react": "^16.9.0 || ^17.0.0 || ^18.0.0"
   },
   "dependencies": {},
   "jest": {

